File: solutions/python/wordy/1/wordy.py
def answer(question):
    if not question.startswith("What is") or not question.endswith("?"):
        raise ValueError("unknown operation")
    
    problem = question.removeprefix("What is").removesuffix("?").strip()
    
    if not problem:
        raise ValueError("syntax error")
    
    tokens = problem.split()
    
    unknown_words = {"cubed", "squared", "what", "who", "where", "when", "why", "how"}
    for word in tokens:
        if word in unknown_words:
            raise ValueError("unknown operation")
    
    def parse_number(token):
        try:
            return int(token)
        except (ValueError, TypeError):
            raise ValueError("syntax error")
    
    result = parse_number(tokens[0])
    i = 1
    
    while i < len(tokens):
        if tokens[i] == "plus":
            if i + 1 >= len(tokens):
                raise ValueError("syntax error")
            result += parse_number(tokens[i + 1])
            i += 2
        elif tokens[i] == "minus":
            if i + 1 >= len(tokens):
                raise ValueError("syntax error")
            result -= parse_number(tokens[i + 1])
            i += 2
        elif tokens[i] == "multiplied" and i + 1 < len(tokens) and tokens[i + 1] == "by":
            if i + 2 >= len(tokens):
                raise ValueError("syntax error")
            result *= parse_number(tokens[i + 2])
            i += 3
        elif tokens[i] == "divided" and i + 1 < len(tokens) and tokens[i + 1] == "by":
            if i + 2 >= len(tokens):
                raise ValueError("syntax error")
            result //= parse_number(tokens[i + 2])
            i += 3
        else:
            try:
                int(tokens[i])
                raise ValueError("syntax error")
            except ValueError as e:
                if "syntax error" in str(e):
                    raise
                raise ValueError("unknown operation")
    
    return result

File: solutions/javascript/wordy/1/wordy.js
export const answer = (question) => {
  if (!question.startsWith('What is') || !question.endsWith('?')) {
    throw new Error('Unknown operation');
  }

  let cleaned = question
    .slice('What is'.length, -1)
    .replace(/multiplied by/g, 'multiplied')
    .replace(/divided by/g, 'divided')
    .trim();

  if (cleaned === '') {
    throw new Error('Syntax error');
  }

  const tokens = cleaned.split(/\s+/);

  const isNumber = (t) => /^-?\d+$/.test(t);
  const operators = new Set(['plus', 'minus', 'multiplied', 'divided']);

if (!isNumber(tokens[0])) {
  if (operators.has(tokens[0])) {
    throw new Error('Syntax error');
  }
  throw new Error('Unknown operation');
}
  let result = parseInt(tokens[0], 10);
  let i = 1;

  while (i < tokens.length) {
    const op = tokens[i];
    if (!operators.has(op)) {
      if (isNumber(op)) {
        throw new Error('Syntax error');
      }
      throw new Error('Unknown operation');
    }
    const next = tokens[i + 1];
    if (next === undefined || !isNumber(next)) {
      throw new Error('Syntax error');
    }
    const num = parseInt(next, 10);

    switch (op) {
      case 'plus':
        result += num;
        break;
      case 'minus':
        result -= num;
        break;
      case 'multiplied':
        result *= num;
        break;
      case 'divided':
        result /= num;
        break;
    }

    i += 2;
  }

  return result;
};